### PR TITLE
Angular upgrade

### PR DESCRIPTION
**Pull request by tchopra91**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,47 +28,50 @@
...
-    "@commitlint/cli": "^8.3.5",
-    "@commitlint/config-conventional": "^8.3.4",
+    "@angular-devkit/build-angular": "~12.2.6",
+    "@angular/cli": "~12.2.6",
+    "@angular/common": "^12.2.6",
+    "@angular/compiler": "^12.2.6",
+    "@angular/compiler-cli": "^12.2.6",
+    "@angular/core": "^12.2.6",
+    "@angular/language-service": "^12.2.6",
+    "@angular/platform-browser": "^12.2.6",
```
Comment (codertushar) [inline package.json:40]: Some of these deps should come as dependencies instead of devDependencies

**Comment by codertushar** [comment]
Also, if you can check if Unit tests are running fine.

**Comment by tchopra91** [comment]
> Also, if you can check if Unit tests are running fine.

Yes, all tests are running fine. :)

File: package.json (hunk 1 above)
Comment (tchopra91) [reply package.json:40]: I can look into this. Also are some changes required under "peerDependencies" when upgrading libraries?

Comment (codertushar) [reply package.json:40]: Yes, we need to update those as well... For now, we can put >=9.x.x for angular for maximum compatibility.
When all our Applications dependent on libs upgrades to angular 12 then we can upgrade peerDependency too to 12.x.x also.

Comment (codertushar) [reply package.json:40]: But I want to know the reason from @kbjerke if there is any particular reason why angular dependencies were put inside devDependencies? I saw on Internet similar being followed in some angular libs(https://github.com/primefaces/primeng/blob/master/package.json) but not in others. I am not aware why is it so.

**Review by codertushar — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (tchopra91) [reply package.json:45]: @kbjerke please find some time to review the changes i did in the last changeset regarding the above discussion and see if they are relevant or if there would be any concerns regarding the same.

Comment (kbjerke) [reply package.json:45]: @codertushar There's no reason for a library to bundle these dependencies, as the project that consumes the library will have those. So you put the required dependencies as peerDeps, and all required to build and package in devDeps

Comment (codertushar) [reply package.json:45]: ok, got your point. Thanks @kbjerke & @tchopra91

**Pull request merged**